### PR TITLE
Fix tox.ini

### PR DESCRIPTION
**Pull request by t-imamichi**
### Summary

Updates tox.ini to avoid the following error due to https://github.com/tox-dev/tox/pull/2671 ([tox 4.0.6](https://github.com/tox-dev/tox/releases/tag/4.0.6) released on Dec 11)

> tox.tox_env.errors.Fail: pass_env values cannot contain whitespace, use comma to have multiple values in a single line, invalid values found 'RAYON_NUM_THREADS OMP_NUM_THREADS QISKIT_PARALLEL RUST_BACKTRACE SETUPTOOLS_ENABLE_FEATURES QISKIT_TESTS QISKIT_IN_PARALLEL'

[log](https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=42213&view=logs&jobId=abfb3e93-d1ff-5b1b-cb4c-74d6dcf536f9&j=abfb3e93-d1ff-5b1b-cb4c-74d6dcf536f9&t=60edc4b3-4259-59c8-ae13-dc31febe2d3d)

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -14,7 +14,7 @@ setenv =
   QISKIT_SUPRESS_PACKAGING_WARNINGS=Y
   QISKIT_TEST_CAPTURE_STREAMS=1
   QISKIT_PARALLEL=FALSE
-passenv = RAYON_NUM_THREADS OMP_NUM_THREADS QISKIT_PARALLEL RUST_BACKTRACE SETUPTOOLS_ENABLE_FEATURES QISKIT_TESTS QISKIT_IN_PARALLEL
+passenv = RAYON_NUM_THREADS, OMP_NUM_THREADS, QISKIT_PARALLEL, RUST_BACKTRACE, SETUPTOOLS_ENABLE_FEATURES, QISKIT_TESTS, QISKIT_IN_PARALLEL
```
Comment (jakelishman) [inline tox.ini:17]: There's still whitespace here between the commas and the starts of the next variables.  Did you mean `RAYON_NUM_THREADS,OMP_NUM_THREADS,...`?

Comment (t-imamichi) [reply tox.ini:17]: I thought whitespace + comma is fine like
https://github.com/Qiskit/qiskit-terra/blob/9c1c18cea4b195749a7c9ed7b278f7a32db63b94/tox.ini#L3
Let's see CI's result.

Comment (jakelishman) [reply tox.ini:17]: CI all passed, so you're almost certainly right.

**Comment by wshanks** [comment]
I came across this while comparing notes on what to do about tox 4 in qiskit-experiments.

If you care about still supporting tox 3, it's better to use newline separation based on [this comment](https://github.com/tox-dev/tox/issues/2676#issuecomment-1345608721). The developers don't particularly encourage supporting tox 3 and 4 together though based on [the following comment](https://github.com/tox-dev/tox/issues/2676#issuecomment-1345609145).

**Pull request merged**